obstore/python/obstore/auth/azure.py

```diff
@@ -0,0 +1,200 @@
+        | azure.identity.UsernamePasswordCredential
+        | azure.identity.VisualStudioCodeCredential
+        | azure.identity.WorkloadIdentityCredential
+    )
+
+    AzureAsyncCredentialUnionType = (
+        azure.identity.aio.AuthorizationCodeCredential
+        | azure.identity.aio.AzureCliCredential
+        | azure.identity.aio.AzureDeveloperCliCredential
+        | azure.identity.aio.AzurePipelinesCredential
+        | azure.identity.aio.AzurePowerShellCredential
+        | azure.identity.aio.CertificateCredential
+        | azure.identity.aio.ChainedTokenCredential
+        | azure.identity.aio.ClientAssertionCredential
+        | azure.identity.aio.ClientSecretCredential
+        | azure.identity.aio.DefaultAzureCredential
+        | azure.identity.aio.EnvironmentCredential
+        | azure.identity.aio.ManagedIdentityCredential
+        | azure.identity.aio.OnBehalfOfCredential
+        | azure.identity.aio.SharedTokenCacheCredential
+        | azure.identity.aio.VisualStudioCodeCredential
+        | azure.identity.aio.WorkloadIdentityCredential
+    )
+
+
+class AzureAuthCredentialProvider:
+    """A CredentialProvider for [AzureStore][obstore.store.AzureStore] that uses [`azure.identity`][].
+
+    This credential provider uses `azure-identity`, and will error if this cannot
+    be imported.
+
+    **Example:**
+
+    ```py
+    from obstore.auth.azure import AzureAuthCredentialProvider
+    from obstore.store import AzureStore
+
+    credential_provider = AzureAuthCredentialProvider(credential=...)
+    store = AzureStore("container", credential_provider=credential_provider)
+    ```
+
+    [`azure.identity`]: https://learn.microsoft.com/en-us/python/api/overview/azure/identity-readme
+    """  # noqa: E501
+
+    credential: AzureCredentialUnionType
+
+    def __init__(
+        self,
+        credential: AzureCredentialUnionType | None = None,
+        scopes: list[str] | None = None,
+        tenant_id: str | None = None,
+    ) -> None:
+        """Create a new AzureAuthCredentialProvider.
+
+        Args:
+            credential: Credential to use for this provider. Defaults to `None`,
+                in which case [`azure.identity.DefaultAzureCredential`][] will be
+                called to find default credentials.
+            scopes: Scopes required by the access token. If not specified,
+                ["https://storage.azure.com/.default"] will be used by default.
+            tenant_id: Optionally specify the Azure Tenant ID which will be passed to
+                the credential's `get_token` method.
+
+        [`azure.identity.DefaultAzureCredential`]: https://learn.microsoft.com/en-us/python/api/azure-identity/azure.identity.defaultazurecredential
+
+        """
+        self.credential = credential or azure.identity.DefaultAzureCredential()
+
+        # Use the Azure Storage scope by default
+        self.scopes = scopes or ["https://storage.azure.com/.default"]
```

Comment:
Can you put this at the top level as
```py
DEFAULT_SCOPES = ("https://storage.azure.com/.default", )
```
(tuple, not list, to be an immutable sequence)

Reply:
Done.